sigstore/oidc.py

```diff
@@ -271,20 +268,6 @@ def __init__(self, base_url: str) -> None:
         except ValueError as exc:
             raise IssuerError(f"OIDC issuer returned invalid configuration: {exc}")
 
-    @classmethod
-    def production(cls) -> Issuer:
-        """
-        Returns an `Issuer` configured against Sigstore's production-level services.
-        """
-        return cls(DEFAULT_OAUTH_ISSUER_URL)
-
-    @classmethod
-    def staging(cls) -> Issuer:
-        """
-        Returns an `Issuer` configured against Sigstore's staging-level services.
-        """
-        return cls(STAGING_OAUTH_ISSUER_URL)
```

Comment:
NB: This is a breaking API change, so we'll need a CHANGELOG entry documenting its removal + to plan it for a new major release (I'm okay with doing that major release whenever 🙂)